app/routes.py

```diff
@@ -130,6 +143,76 @@ def transpile_circuit():
                     'transpiled-qasm': transpiled_circuit.qasm()}), 200
 
 
+@app.route('/qiskit-service/api/v1.0/analyze-original-circuit', methods=['POST'])
+def analyze_original_circuit():
+
+    if not request.json:
+        abort(400)
+
+    impl_language = request.json.get('impl-language', '')
+    impl_url = request.json.get('impl-url', "")
+    input_params = request.json.get('input-params', "")
+    bearer_token = request.json.get("bearer-token", "")
+    input_params = parameters.ParameterDictionary(input_params)
+
+    if impl_url is not None and impl_url != "":
+        impl_url = request.json['impl-url']
+        if impl_language.lower() == 'openqasm':
+            short_impl_name = 'no name'
+            circuit = implementation_handler.prepare_code_from_qasm_url(impl_url, bearer_token)
+        else:
+            short_impl_name = "untitled"
+            try:
+                circuit = implementation_handler.prepare_code_from_url(impl_url, input_params, bearer_token)
+            except ValueError:
+                abort(400)
+
+    elif 'impl-data' in request.json:
+        impl_data = base64.b64decode(request.json.get('impl-data').encode()).decode()
+        short_impl_name = 'no short name'
+        if impl_language.lower() == 'openqasm':
+            circuit = implementation_handler.prepare_code_from_qasm(impl_data)
+        else:
+            try:
+                circuit = implementation_handler.prepare_code_from_data(impl_data, input_params)
+            except ValueError:
+                abort(400)
```

Comment:
@aaronschi could you enable to also support `qasm-string` as input, see here https://github.com/UST-QuAntiL/qiskit-service/blob/491223280db44dc38ac3dfda8a6bd33679a15b3f/app/routes.py#L86